R-packages/covidcast/R/plot.R

```diff
@@ -346,22 +350,32 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
   geom_args$data = map_df
   polygon_layer = do.call(ggplot2::geom_polygon, geom_args)
 
-  # Set the lats and lons for counties
+  # Retrieve coordinates for mapping
+  # Reading from usmap files to ensure consistency with borders
   if (attributes(x)$geo_type == "county") {
-    g = county_geo[county_geo$FIPS %in% map_geo, ]
-    cur_geo = g$FIPS
-    cur_lon = g$LON
-    cur_lat = g$LAT
+    centroid_file = system.file("extdata",
+                                "us_counties_centroids.csv",
+                                package = "usmap")
+    centroids = utils::read.csv(centroid_file,
+                                colClasses = centroid_col_classes)
+    centroids = centroids[centroids$fips %in% map_geo, ]
+
+    # Ordering must match due to plotting code
+    centroids = centroids[match(map_geo, centroids$fips), ]
+    cur_geo = centroids$fips
     cur_val = rep(NA, length(cur_geo))
   }
-
-  # Set the lats and lons for states
   else if (attributes(x)$geo_type == "state") {
-    state_geo$STATE = tolower(state_geo$STATE)
-    g = state_geo[state_geo$STATE %in% map_geo, ]
-    cur_geo = g$STATE
-    cur_lon = g$LON
-    cur_lat = g$LAT
+    centroid_file = system.file("extdata",
+                                "us_states_centroids.csv",
+                                package = "usmap")
+    centroids = utils::read.csv(centroid_file,
+                                colClasses = centroid_col_classes)
+    centroids$abbr = tolower(centroids$abbr)
+    centroids = centroids[centroids$abbr %in% map_geo, ]
+    centroids = centroids[match(map_geo, centroids$abbr), ]
```

Comment:
What does this do?

Reply:
The first line is the equivalent of the previous `g = county_geo[county_geo$FIPS %in% map_geo, ]` and reduces the dataframe to the counties that are being mapped.

The 2nd line is superfluous and has been removed. I had an issue with data being mapped in the wrong spot and this was a fix. However, the root cause had to do with loading the fips column as numeric instead of character so it's no longer necessary after I fixed that.